packages/@ember/-internals/runtime/lib/ext/function.js

```diff
@@ -77,6 +79,14 @@ if (ENV.EXTEND_PROTOTYPES.Function) {
       enumerable: false,
       writable: true,
       value: function() {
+        deprecate(
+          `Function prototype extensions have been deprecated, please migrate from function(){}.property('bar') to computed('bar', function() {}).`,
+          false,
+          {
+            id: 'ember-runtime.function.property',
+            until: '4.0.0',
```

Comment:
Do we also need `since: '3.10.0'` (or `'3.11.0'`) here?

Reply:
No, `deprecate`'s options do not include a `since` field. The intent is for the `@ember/deprecated-features` exports to do that, and ultimately be used to enable svelte infrastructure.